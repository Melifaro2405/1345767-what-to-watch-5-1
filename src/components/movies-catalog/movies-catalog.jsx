import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {filmProptypes} from "../../props-validation";
import MoviesList from "../movies-list/movies-list";
import GenresList from "../genres-list/genres-list";
import ButtonShowMore from "../button-show-more/button-show-more";
import {getFilmsByGenre} from "../../store/selectors/selectors";

const MoviesCatalog = ({filteredFilms, countShownFilms}) => {
  return (
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>
      <GenresList />
      <MoviesList films={filteredFilms.slice(0, countShownFilms)} />
      {(filteredFilms.length >= countShownFilms) ? <ButtonShowMore /> : null}
    </section>
  );
};

MoviesCatalog.propTypes = {
  countShownFilms: PropTypes.number.isRequired,
  filteredFilms: PropTypes.arrayOf(PropTypes.shape(filmProptypes)).isRequired
};

const mapStateToProps = ({DATA, APP_STATE}) => ({
  countShownFilms: APP_STATE.countShownFilms,
  filteredFilms: getFilmsByGenre({DATA, APP_STATE})
});

export {MoviesCatalog};
export default connect(mapStateToProps)(MoviesCatalog);
