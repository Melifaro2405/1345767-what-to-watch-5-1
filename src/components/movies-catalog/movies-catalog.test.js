import React from 'react';
import renderer from 'react-test-renderer';
import {MoviesCatalog} from './movies-catalog';

jest.mock(`../genres-list/genres-list`, () => `GenresList`);
jest.mock(`../movies-list/movies-list`, () => `MoviesList`);
jest.mock(`../button-show-more/button-show-more`, () => `ButtonShowMore`);

const CountFilms = {
  LESS_DISPLAYED: 1,
  MORE_DISPLAYED: 8
};

const films = [
  {
    id: 1,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 2,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 3,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  }
];

describe(`Should MoviesCatalog render correctly`, () => {
  test.each([
    [`with`, CountFilms.LESS_DISPLAYED],
    [`without`, CountFilms.MORE_DISPLAYED],
  ])(`%s buttonShowMore`, (_expected, countShownFilms) => {
    const tree = renderer
      .create(<MoviesCatalog
        filteredFilms={films}
        countShownFilms={countShownFilms}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
