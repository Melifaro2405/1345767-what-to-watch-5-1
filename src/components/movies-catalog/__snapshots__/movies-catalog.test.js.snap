// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MoviesCatalog render correctly with buttonShowMore 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <GenresList />
  <MoviesList
    films={
      Array [
        Object {
          "details": Object {
            "runtime": 99,
          },
          "id": 1,
          "moreInfo": Object {
            "backgroundColor": "#ffffff",
            "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
            "genre": "Comedy",
            "isAddToMyList": false,
            "playPreviewSrc": "https://some-link",
            "playVideoSrc": "https://some-link",
            "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
            "releaseDate": 2014,
          },
          "overview": Object {
            "actorsList": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "rating": 8.9,
            "ratingCount": 240,
          },
          "preview": Object {
            "src": "img/the-grand-budapest-hotel-poster.jpg",
            "title": "The Grand Budapest Hotel",
          },
        },
      ]
    }
  />
  <ButtonShowMore />
</section>
`;

exports[`Should MoviesCatalog render correctly without buttonShowMore 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <GenresList />
  <MoviesList
    films={
      Array [
        Object {
          "details": Object {
            "runtime": 99,
          },
          "id": 1,
          "moreInfo": Object {
            "backgroundColor": "#ffffff",
            "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
            "genre": "Comedy",
            "isAddToMyList": false,
            "playPreviewSrc": "https://some-link",
            "playVideoSrc": "https://some-link",
            "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
            "releaseDate": 2014,
          },
          "overview": Object {
            "actorsList": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "rating": 8.9,
            "ratingCount": 240,
          },
          "preview": Object {
            "src": "img/the-grand-budapest-hotel-poster.jpg",
            "title": "The Grand Budapest Hotel",
          },
        },
        Object {
          "details": Object {
            "runtime": 99,
          },
          "id": 2,
          "moreInfo": Object {
            "backgroundColor": "#ffffff",
            "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
            "genre": "Comedy",
            "isAddToMyList": false,
            "playPreviewSrc": "https://some-link",
            "playVideoSrc": "https://some-link",
            "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
            "releaseDate": 2014,
          },
          "overview": Object {
            "actorsList": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "rating": 8.9,
            "ratingCount": 240,
          },
          "preview": Object {
            "src": "img/the-grand-budapest-hotel-poster.jpg",
            "title": "The Grand Budapest Hotel",
          },
        },
        Object {
          "details": Object {
            "runtime": 99,
          },
          "id": 3,
          "moreInfo": Object {
            "backgroundColor": "#ffffff",
            "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
            "genre": "Comedy",
            "isAddToMyList": false,
            "playPreviewSrc": "https://some-link",
            "playVideoSrc": "https://some-link",
            "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
            "releaseDate": 2014,
          },
          "overview": Object {
            "actorsList": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "rating": 8.9,
            "ratingCount": 240,
          },
          "preview": Object {
            "src": "img/the-grand-budapest-hotel-poster.jpg",
            "title": "The Grand Budapest Hotel",
          },
        },
      ]
    }
  />
</section>
`;
